src/layers/TracesLayer.ts

```diff
@@ -0,0 +1,40 @@
+import { Lines } from "@/Chart";
+import { LayerArgs, LayerType, OptionalLayer } from "./Layer";
+
+export class TracesLayer extends OptionalLayer {
+  type = LayerType.Trace;
+
+  constructor(public lines: Lines) {
+    super();
+  };
+
+  draw = (layerArgs: LayerArgs) => {
+    const subLayers = this.lines.map((l, index) => {
+      const linePath = layerArgs.scaleConfig.lineGen(l.points);
+      return layerArgs.coreLayers[LayerType.BaseLayer].append("path")
+        .attr("id", `${layerArgs.getHtmlId(LayerType.Trace)}-${index}`)
+        .attr("pointer-events", "none")
+        .attr("vector-effect", "non-scaling-stroke")
+        .attr("fill", "none")
+        .attr("stroke", l.style.color || "black")
+        .attr("opacity", l.style.opacity || 1)
```

Comment:
I'm slightly surprised there isn't already a type defining the styles you can apply in a d3 element, but there doesn't seem to be!